File: companies/mujin/joint_point.py
def digits_sum(n: int) -> int:
    """Compute the sum of digits in the number."""
    return sum(int(d) for d in str(n))


def next_num(n: int) -> int:
    """Find the number in the sequence."""
    return n + digits_sum(n)


def find_joint_point(s1: int, s2: int) -> int:
    """Find the joint point of two sequences starting from s1 and s2."""
    while s1 != s2:
        if s1 < s2:
            s1 = next_num(s1)
        else:
            s2 = next_num(s2)

    return s1


if __name__ == '__main__':
    s1 = 34
    s2 = 40
    print(f"Joint point of {s1} and {s2}: {find_joint_point(s1, s2)}")

File: companies/mujin/joint_point.cpp
#include <iostream>
#include <string>

using namespace std;


int digitSum(int n) {
    /*Compute the sum of digits in the number.*/
    string num = to_string(n);
    int sum = 0;
    for (auto& d : num) {
        sum += d - '0';
    }
    return sum;
}


int nextNum(int n) {
    /*Find the next number in the sequence.*/
    return n + digitSum(n);
}


int findJointPoint(int s1, int s2) {
    /*Find the joint point of two sequences starting from s1 and s2.*/
    while (s1 != s2) {
        if (s1 < s2) {
            s1 = nextNum(s1);
        } else {
            s2 = nextNum(s2);
        }
    }
    return s1;
}

int main(int argc, char **argv) {
    int s1 = 34;
    int s2 = 40;
    cout << "Joint point of " << s1 << " and " << s2 << ": " << findJointPoint(s1, s2) << endl;
}